src/memory_pool_internal.h

```diff
@@ -29,6 +31,9 @@ typedef struct umf_memory_pool_t {
     umf_memory_provider_handle_t provider;
     // Tells whether memory provider is owned by the pool.
     bool own_provider;
+
+    // linear base allocator used by the pool
+    umf_ba_linear_pool_t *pool_linear;
```

Comment:
Does it really make sense to use a separate allocator for each pool? this means each pool now has 4k metadata overhead. Wouldn't it be better to just use a global base_alloc instance for all umf_memory_pool_t allocations? Similarly for umf_memory_provider_t.

Reply:
What and when would destroy such a global instance of base allocator? Do you want to add also a global reference counter? Will not it be too complicated? Is it worth that?

Reply:
@igchor Done. Did you mean that?

Reply:
I meant that using base_alloc (i.e. umf_ba_alloc / umf_ba_free) would probably be better vs using a linear allocator. For linear allocator per pool/provider we waste ~4K but even for global instance there is a problem with freeing the pool/provider metadata (basically we never free, before the library is destroyed).

Why can't we have a single global base_alloc pool (not linear allocator) for a size (let's say 64B) that would be big enough to hold metadata for both memory_pool_t, memory_provider_t and memory_tracking_provider_t? There would still be some waste but it would be pretty low (all those structs are are of similar size I think) and we can free the metadata at any time.

Reply:
@igchor Done. Did you mean that?

Reply:
Yes, exactly. And I even think we can do the same for memspace/memory_targets (as they can be created/destroyed at any time as well, sorry I forgot about them earlier).

Reply:
@ldorau @igchor - this code is getting more and more complicated. If we use base allocator for different sizes then why just we don't use linear allocator everywhere?

Reply:
Done